src/Utilities/dense_dict.jl

```diff
@@ -22,15 +22,16 @@ end
 
 # Implementation of the `AbstractDict` API.
 # Base.empty(::DenseDict, ::Type{K}, ::Type{V}) not implemented
-function Base.iterate(d::DenseDict, args...)
+function Base.iterate(d::DenseDict{K,V}, args...) where {K,V}
     itr = iterate(d.set, args...)
     if itr === nothing
         return nothing
     else
         el, i = itr
-        return d.inverse_hash(el) => d.map[el], i
+        return d.inverse_hash(el)::K => d.map[el]::V, i
```

Comment:
Are the `::K` and `::V` needed ?

Reply:
Just tested, Julia does not seem to infer those correctly. Therefore the is a very minor difference a little less than 1%.
I vote for leaving it there, since its not restricting anything.
However, if we really want to avoid that, I can remove.

Reply:
No if Julia inference is not doing it correctly then leaving it there is what we should do